src/core_ct/core.py

```diff
@@ -106,7 +106,131 @@ def trim(self, axis: int, loc_start: int, loc_end: int | None = None) -> None:
                     :, :, loc_start : len(self.pixel_array[0, 0]) - loc_end
                 ]
             case _:
-                raise Exception("axis must be a value between 0 and 2 (inclusive)")
+                raise ValueError("axis must be a value between 0 and 2 (inclusive)")
+
+    def swapaxes(self, axis1: int, axis2: int) -> None:
+        """
+        Swap two axes of pixel_array and update pixel_dimensions accordingly.
+
+        Arguments:
+        ---------
+            axis1: integer either 0,1,2 specifying one axis to swap:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+            axis2: integer either 0,1,2 specifying the second axis to swap:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+
+        Returns:
+        -------
+            None
```

Comment:
I'm not sure if we made a decision on this, but if we are changing functions like `trim` to make a copy of the core rather than operating in-place, should we also have `rotate`, `flip`, and `swapaxes` also make a copy of a core for consistency? It feels a little strange to me though to have these functions not in-place.

Reply:
Updated them to be not in-place